dashboard/test/controllers/aichat_controller_test.rb

```diff
@@ -236,64 +160,36 @@ class AichatControllerTest < ActionController::TestCase
     assert_equal chat_event2_response["chatMessageText"], chat_event2_stored["chatMessageText"]
   end
 
-  # chat_request tests
-  test 'GET chat_request returns not found if request does not exist' do
-    sign_in(@authorized_student1)
-    get :chat_request, params: {id: 100}, as: :json
-    assert_response :not_found
-  end
-
-  test 'GET chat_request returns forbidden if user is not the requester' do
-    sign_in(@authorized_teacher1)
-    request = create(:aichat_request, user: @authorized_student1)
-    get :chat_request, params: {id: request.id}, as: :json
-    assert_response :forbidden
-  end
-
-  test 'GET chat_request returns request status and response' do
-    response = "AI model response"
-    execution_status = SharedConstants::AI_REQUEST_EXECUTION_STATUS[:SUCCESS]
-
-    sign_in(@authorized_teacher1)
-    request = create(:aichat_request, user: @authorized_teacher1, response: response, execution_status: execution_status)
-    get :chat_request, params: {id: request.id}, as: :json
-
-    assert_response :success
-    assert_equal json_response.keys, ['executionStatus', 'response']
-    assert_equal json_response['executionStatus'], execution_status
-    assert_equal json_response['response'], response
-  end
-
   # user_has_access tests
   test 'signed out user does not have access to user_has_access test' do
-    get :user_has_access
+    get :user_has_access, format: :json
```

Comment:
I had to add these so that we would get forbidden responses if you're signed out, rather than redirected to sign in (which is what happens if our existing code that uses this API were to hit this as a signed out user). This little bit is responsible as I understand it:

https://github.com/code-dot-org/code-dot-org/blob/3721c7e8d6f7c8c1813d8f8443c4c060ecb1b363/dashboard/app/controllers/application_controller.rb#L68-L78